### ISSUE TITLE
代码咨询

### ISSUE THREAD

**Issue by fengxiangbo1** [YR0211/MulFS-CAP#2]
您好我看您文章中说训练的batchsize是4，但是目前给出的代码是1，当设置为4时，代码会显示通道维数错误，我对代码进行分批次处理以后，代码没有报错，但一致没有进行训练进度，我想请问，你后续是如何处理的

**Comment by esportsxi** [comment]
这位老兄，我正常修改完之后是有训练进度的，你可以试试先把训练集图片设置成10张，然后跑一下看看，每一个epoch还挺久的。我一开始也看着半天不动，然后看GPU的占用是有的。但是我也有个问题，训练过程保存的图片checkpoint\images里面的都是灰度图，没有彩色图片，不知道咋搞

**Comment by pcc-99** [comment]
> 这位老兄，我正常修改完之后是有训练进度的，你可以试试先把训练集图片设置成10张，然后跑一下看看，每一个epoch还挺久的。我一开始也看着半天不动，然后看GPU的占用是有的。但是我也有个问题，训练过程保存的图片checkpoint\images里面的都是灰度图，没有彩色图片，不知道咋搞

老兄，你的预训练模型是怎么处理的

**Comment by lyy1900** [comment]
> 您好我看您文章中说训练的batchsize是4，但是目前给出的代码是1，当设置为4时，代码会显示通道维数错误，我对代码进行分批次处理以后，代码没有报错，但一致没有进行训练进度，我想请问，你后续是如何处理的

你好，请问如何解决修改batchsize为4后，代码显示的通道维度错误

**Comment by esportsxi** [comment]
> > 这位老兄，我正常修改完之后是有训练进度的，你可以试试先把训练集图片设置成10张，然后跑一下看看，每一个epoch还挺久的。我一开始也看着半天不动，然后看GPU的占用是有的。但是我也有个问题，训练过程保存的图片checkpoint\images里面的都是灰度图，没有彩色图片，不知道咋搞
> 
> 老兄，你的预训练模型是怎么处理的

是指处理什么，我直接跑的train.py

**Comment by esportsxi** [comment]
> > 您好我看您文章中说训练的batchsize是4，但是目前给出的代码是1，当设置为4时，代码会显示通道维数错误，我对代码进行分批次处理以后，代码没有报错，但一致没有进行训练进度，我想请问，你后续是如何处理的
> 
> 你好，请问如何解决修改batchsize为4后，代码显示的通道维度错误

这是在保存图片的时候出现的报错，你改一下utils中的save_img部分的代码就行

**Comment by lzl983487562** [comment]
> > > 您好我看您文章中说训练的batchsize是4，但是目前给出的代码是1，当设置为4时，代码会显示通道维数错误，我对代码进行分批次处理以后，代码没有报错，但一致没有进行训练进度，我想请问，你后续是如何处理的
> > 
> > 
> > 你好，请问如何解决修改batchsize为4后，代码显示的通道维度错误
> 
> 这是在保存图片的时候出现的报错，你改一下utils中的save_img部分的代码就行

请问一下如何进行修改呢？

**Comment by lyy1900** [comment]
> > > > 您好我看您文章中说训练的batchsize是4，但是目前给出的代码是1，当设置为4时，代码会显示通道维数错误，我对代码进行分批次处理以后，代码没有报错，但一致没有进行训练进度，我想请问，你后续是如何处理的
> > > 
> > > 
> > > 你好，请问如何解决修改batchsize为4后，代码显示的通道维度错误
> > 
> > 
> > 这是在保存图片的时候出现的报错，你改一下utils中的save_img部分的代码就行
> 
> 请问一下如何进行修改呢？

我尝试进行了如下修改，但无法保证正确性：
def save_img(x, save_dir):
    # 训练的时候设置batch size为4
    # x = x.squeeze(dim=0).permute(1, 2, 0)

    x = x[0]  # 获取第 idx 张图像
    x = x.permute(1, 2, 0)  # 转换为 (H, W, C)

    x = x.cpu().detach().numpy()
    x = (x * 255.).astype(np.float64)
    cv2.imwrite(save_dir, x)

**Comment by gg680** [comment]
> 这位老兄，我正常修改完之后是有训练进度的，你可以试试先把训练集图片设置成10张，然后跑一下看看，每一个epoch还挺久的。我一开始也看着半天不动，然后看GPU的占用是有的。但是我也有个问题，训练过程保存的图片checkpoint\images里面的都是灰度图，没有彩色图片，不知道咋搞

您好 ，请问您解决了这个彩色图片问题吗，我更改了部分代码 但是效果很差
